### DISCUSSION TITLE
Refactor our renovate docker images

### DISCUSSION THREAD

**Discussion by viceice** [renovatebot/renovate#23433] [Suggest an Idea]
### Tell us more.

Basic idea is to create a slim and a full base image and add renovate on top on it on renovate releases.
This will cause much faster docker image releases and we can probably use it for local testing too.

## Steps
- create a new `docker-base` repo to create the slim and full base images pushed as `ghcr.io/renovatebot/base`
- add a conditional `dockerfile` to add renovate an build the image after the npm release to this repo

## Related issues
- #23377

**Comment by lmilbaum** [comment]
Let me see that I understand the proposal.
The idea, you're suggesting, is generating an image which is everything renovate needs without renovate.
Then, this image can be used for generating the renovate image releases but also used in renovate functional tests from PRs in the renovate repo?

**Reply by viceice** [reply]
yes, i don't like to always need to build the full image when a new renovate release is created.

One drawback is, we no longer update the latest renovate image with newer tools.

**Reply by rarkins** [reply]
I think we found stagger tool updating to be less frequent (Eg weekly) if it means better efficiency of building and downloading renovate images

**Reply by lmilbaum** [reply]
I like this approach. @viceice I don't see the drawback you have mentioned. The renovate container image build can be triggered by having a new base image version or/and a new renovate release.
